src/meta/processors/admin/BalanceTask.cpp

```diff
@@ -0,0 +1,234 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/processors/admin/BalanceTask.h"
+#include <folly/synchronization/Baton.h>
+#include "meta/processors/Common.h"
+
+namespace nebula {
+namespace meta {
+
+#define SAVE_STATE() \
+    if (!saveInStore()) { \
+        ret_ = Result::FAILED; \
+        onError_(); \
+        return; \
+    }
+
+const std::string kBalanceTaskTable = "__b_task__"; // NOLINT
+
+void BalanceTask::invoke() {
+    CHECK_NOTNULL(client_);
+    if (ret_ == Result::FAILED) {
+        endTimeMs_ = time::TimeUtils::nowInSeconds();
+        saveInStore();
+        onError_();
+        return;
+    }
+    switch (status_) {
+        case Status::START: {
+            LOG(INFO) << taskIdStr_ << "Start to move part!";
+            status_ = Status::CHANGE_LEADER;
+            ret_ = Result::IN_PROGRESS;
+            startTimeMs_ = time::TimeUtils::nowInSeconds();
+        }
+        case Status::CHANGE_LEADER: {
+            LOG(INFO) << taskIdStr_ << "Ask the src to give up the leadership.";
+            SAVE_STATE();
+            client_->transLeader(spaceId_, partId_, src_, dst_).thenValue([this](auto&& resp) {
+                if (!resp.ok()) {
+                    ret_ = Result::FAILED;
```

Comment:
How about sleep a while if failed or retry only a few times, otherwise it would invoke all the times?

Reply:
Good suggestion!  I will wrap retry logic inside AdminClient.